token-service/src/main/java/org/p2p/token/service/database/TokenPriceDao.kt

```diff
@@ -20,10 +20,10 @@ internal interface TokenPriceDao {
     suspend fun getAllTokensPrices(): List<TokenServicePriceEntity>
 
     @Query("SELECT * FROM token_price_table")
-    fun getAllTokensPricesFlow(): Flow<TokenServicePriceEntity>
+    fun getAllTokensPricesFlow(): Flow<List<TokenServicePriceEntity>>
 
-    @Query("SELECT * FROM token_price_table WHERE token_address = :tokenAddress")
-    suspend fun getTokenPriceByAddress(tokenAddress: String): TokenServicePriceEntity?
+    @Query("SELECT * FROM token_price_table WHERE token_column_key = :tokenColumnKey")
```

Comment:
why the previous logic was not working?

Reply:
we have collision with native eth and native sol tokens rates